[chore] Change receiver config tests to unmarshal config only for that component. (Part 2)

kovrus opened the pull request:
**Description:** <Describe what has changed.>
<!--Ex. Fixing a bug - Describe the bug and how this fixes the issue.
Ex. Adding a feature - Explain what this achieves.-->
Change receivers config tests to unmarshal config only for that component.
Part 1 - https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/13383

- [x] hostmetricsreceiver
- [x] iisreceiver
- [x] influxdbreceiver
- [x] jmxreceiver
- [x] journaldreceiver
- [x] k8sclusterreceiver
- [x] k8seventsreceiver
- [x] kafkametricsreceiver
- [x] kafkareceiver
- [x] kubeletstatsreceiver
- [x] prometheusexecreceive

**Link to tracking Issue:** <Issue number if applicable>
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13224

jpkrohling commented on receiver/jaegerreceiver/testdata/bad_empty_config.yaml:
```diff
@@ -1,15 +0,0 @@
-receivers:
```
Can you confirm the removal of those files are intended?

kovrus replied:
yes, I merged them into https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14476/files/1db4f14070ab8b86c65ddb8d8d161fe68d827a15#diff-56f282193f5228abfb85e5e670e12b201672b82d126baa296a01268e65f6954cR54

dmitryax commented:
I would like to help with reviewing it but it's huge :) Any change it can be broken down into smaller PRs, like 5 of them?

kovrus committed:
[chore] Change receiver config tests to unmarshal config only for that component. (part2)

- [x] hostmetricsreceiver
- [x] iisreceiver
- [x] influxdbreceiver
- [x] jmxreceiver
- [x] journaldreceiver
- [x] k8sclusterreceiver
- [x] k8seventsreceiver
- [x] kafkametricsreceiver
- [x] kafkareceiver
- [x] kubeletstatsreceiver
- [x] prometheusexecreceiver

kovrus commented:
> I would like to help with reviewing it but it's huge :) Any chance it can be broken down into smaller PRs, like 5 of them?

@dmitryax split it into 4 PRs, which should be fine since the changes are pretty straightforward:

https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14476
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14561
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14562
https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14563
